lib/r10k/action/deploy/environment.rb

```diff
@@ -113,6 +113,10 @@ def visit_deployment(deployment)
 
           if @settings.dig(:overrides, :purging, :purge_levels).include?(:deployment)
             logger.debug("Purging unmanaged environments for deployment...")
+            deployment.sources.each do |source|
+              source.cache.sync!
+              source.instance_variable_set(:@environments, [])
```

Comment:
What is the purpose of this?

Reply:
I was actually wondering if it wouldn't be better to add a method to https://github.com/puppetlabs/r10k/blob/main/lib/r10k/source/git.rb to allow resetting the environments "officially", instead of going through Ruby's hoops.

Reply:
ahh. yeah, I think adding a method would be good. I don't really have anything to back this up, but `instance_variable_set` feels like a code smell to me?

Reply:
That sounds good to me, whatever we'd call that method on source or deployment or whatever should probably either be a no-op on non-VCS backed Environments, or conditionally called.